### PR TITLE
newEnforcer() function added

### PR DESCRIPTION
**Pull request by NitinBhasneria**
This PR has added the newEnforcer function. Also, newEnforcer function uses some more function inside, as they are not defined I have commented those functions call. This will be included in the function when defined.
Thanks

### CODE REVIEW

**Comment by NitinBhasneria** [comment]
@techoner @hsluoyz @PsiACE please review. Thanks

**Comment by NitinBhasneria** [comment]
@xcaptian can you review my pr. Thanks

**Comment by xcaptain** [comment]
@NitinBhasneria Thanks for the contribution, currently we still don't have a MVP, so I would encourage people to do more work and get a more complete library then submit the pull request.

I think @DivyPatel9881 has done a lot of work at https://github.com/casbin/casbin-cpp/pull/16 though still not a MVP. Maybe we could do some competitions like a hackathon and then dicide whose branch would be merged into the official repo.

**Comment by NitinBhasneria** [comment]
@xcaptain I got your point but I also pushed 4 more PR's they also contain my work. Also, we have given the guideline for not pushing more than 100 LOC. In Gitter too @hsluoyz told us to open second pr on the basis on first while the second pr does not contain the first PR's code.

File: casbin/enforcer.cpp [hunk 1]
```diff
@@ -20,3 +20,113 @@
...
+    if (paramLen >= 1) {
+        if(params[paramLen-1] == 'b')
+		{
+		    // e->EnableLog(enableLog);
+
+			parsedParamLen++;
+		}
+	}
+
+    if( paramLen-parsedParamLen == 2) {
```
Comment (xcaptain) [inline casbin/enforcer.cpp]: No need to do these type checkings, use overload would much simpler.

**Comment by xcaptain** [comment]
图片

I don't know if minium working code means minium viable product, and I also agree that we shouldn't make too large pull request, but I'm afraid your branch would have a lot of conflict with other people's if we merge now.

@hsluoyz What do you think about the very early stage contribution?

File: casbin/enforcer.cpp [hunk 2]
```diff
@@ -20,3 +20,113 @@
...
+//
+// MySQL DB:
+//
+// 	a := mysqladapter.NewDBAdapter("mysql", "mysql_username:mysql_password@tcp(127.0.0.1:3306)/");
+// 	string error;
+//  Enforcer e;
+//  tie(e, error)= casbin.NewEnforcer("path/to/basic_model.conf", a);
+//
+template<typename R,typename ... T>
+tuple <Enforcer, string> newEnforcer(R modelPath, T ... Q ){
```
[the remark this answers is not in the document]
Comment (hsluoyz) [reply casbin/enforcer.cpp]: Yes. We will use constructor overload because C++ supports it.

**Comment by hsluoyz** [comment]
This PR can be closed. Please follow my guide.

**Comment by NitinBhasneria** [comment]
@hsluoyz according to your guide we should not push more than 100LoC right?

**Pull request closed without merging**